DQM/HcalTasks/plugins/ZDCQIE10Task.cc

```diff
@@ -1,116 +1,579 @@
     _cADC_vs_TS_EChannel[didm()]->setAxisTitle("TS", 1);
     _cADC_vs_TS_EChannel[didm()]->setAxisTitle("sum ADC", 2);
+
+    ib.setCurrentFolder("Hcal/ZDCQIE10Task/fC_perChannel");
+    _cfC_EChannel[didm()] = ib.book1DD(histoname.c_str(), histoname.c_str(), 100, 0, 8000);
+    _cfC_EChannel[didm()]->setAxisTitle("fC", 1);
+    _cfC_EChannel[didm()]->setAxisTitle("N", 2);
+    ib.setCurrentFolder("Hcal/ZDCQIE10Task/fC_vs_TS_perChannel");
+    _cfC_vs_TS_EChannel[didm()] = ib.book1DD(histoname.c_str(), histoname.c_str(), 6, 0, 6);
+    _cfC_vs_TS_EChannel[didm()]->setAxisTitle("TS", 1);
+    _cfC_vs_TS_EChannel[didm()]->setAxisTitle("sum fC", 2);
+
+    histoname = "RPD_M_" + std::to_string(channel);
+    _cTDC_EChannel[didm()] = ib.book1DD(histoname.c_str(), histoname.c_str(), 50, 1, 50);
+    _cTDC_EChannel[didm()]->setAxisTitle("TDC", 1);
+    _cTDC_EChannel[didm()]->setAxisTitle("N", 2);
   }
 }
 
-/* virtual */ void ZDCQIE10Task::analyze(edm::Event const& e, edm::EventSetup const&) {
+/* virtual */
+void ZDCQIE10Task::_process(edm::Event const& e, edm::EventSetup const& es) {
+  const HcalTopology& htopo = es.getData(htopoToken_);
+  const HcalLongRecoParams& p = es.getData(paramsToken_);
+  longRecoParams_ = std::make_unique<HcalLongRecoParams>(p);
+  longRecoParams_->setTopo(&htopo);
+
+  std::vector<unsigned int> mySignalTS;
+  std::vector<unsigned int> myNoiseTS;
+
+  int bx = e.bunchCrossing();
+
+  edm::Handle<BXVector<l1t::EtSum> > sums;
+  e.getByToken(sumToken_, sums);
+
+  //int startBX = sums->getFirstBX();
+
+  //for (int ibx = startBX; ibx <= sums->getLastBX(); ++ibx){
+  //	for (auto itr = sums->begin(ibx); itr != sums->end(ibx); ++itr){
+  //		if (itr->getType() == l1t::EtSum::EtSumType::kZDCP)
+  //			cout << "Plus side!" << endl;
+  //		if (itr->getType() == l1t::EtSum::EtSumType::kZDCM)
+  //			cout << "Minus side!" << endl;
+  //	}
+  //}
+
   edm::Handle<QIE10DigiCollection> digis;
   if (!e.getByToken(_tokQIE10, digis))
     edm::LogError("Collection QIE10DigiCollection for ZDC isn't available" + _tagQIE10.label() + " " +
                   _tagQIE10.instance());
 
+  double HADM_sum = 0, HADP_sum = 0, EMM_sum = 0, EMP_sum = 0;
+  double HADM_tot_sum = 0, HADP_tot_sum = 0, EMM_tot_sum = 0, EMP_tot_sum = 0;
+
   for (auto it = digis->begin(); it != digis->end(); it++) {
     const QIE10DataFrame digi = static_cast<const QIE10DataFrame>(*it);
+
     HcalZDCDetId const& did = digi.detid();
 
+    HcalZDCDetId cell = it->id();
+    DetId detcell = (DetId)cell;
+
+    const HcalLongRecoParam* myParams = longRecoParams_->getValues(detcell);
+    mySignalTS.clear();
+    myNoiseTS.clear();
+    mySignalTS = myParams->signalTS();
+    myNoiseTS = myParams->noiseTS();
+
+    for (unsigned int k : mySignalTS) {
+      std::cout << k << ' ';
+    }
+
+    //cout << " ZDC did" <<did()<<endl;
+    uint32_t rawid = _ehashmap.lookup(did);
+
+    if (rawid == 0) {
```

Comment:
There are some empty if statements in this code, unless it is being used for something we should remove.